bot/cogs/filtering.py

```diff
@@ -159,6 +185,21 @@ def mod_log(self) -> ModLog:
                         triggered = await _filter["function"](msg)
 
                     if triggered:
+                        # If the message is classed as offensive, we store it in the site db and
+                        # it will be deleted it after one week.
+                        if _filter["offensive_msg"]:
+                            delete_date = msg.created_at.date() + OFFENSIVE_MSG_DELETE_TIME
+                            await self.bot.api_client.post(
+                                'bot/offensive-message',
+                                json={
+                                    'id': msg.id,
+                                    'channel_id': msg.channel.id,
+                                    'delete_date': delete_date.isoformat()
+                                }
+                            )
+                            log.trace(f"Offensive message will be deleted on "
+                                      f"{delete_date.isoformat()}")
```

Comment:
This should go after the next logic block to remove the possibility that we send a message to the API that the filter has already deleted.